lec04-dplyr.Rmd

```diff
@@ -1,4 +1,724 @@
+the aesthetic function to include `group = species_id`:
+
+```{r time-series-by-species}
+surveys_abun_species %>%
+    group_by(year, species) %>%
+    tally() %>% 
+    ggplot(aes(x = year, y = n, group = species)) +
+        geom_line()
+```
+
+We will be able to distinguish species in the plot if we add colors (using `color` also automatically groups the data):
+
+```{r time-series-with-colors}
+surveys_abun_species %>%
+    group_by(year, species) %>%
+    tally() %>% 
+    ggplot(aes(x = year, y = n, color = species)) + # `color` groups automatically
+        geom_line() # try adding `size = 2`
+```
+
+### Faceting
+
+ggplot has a special technique called *faceting* that allows the user to split one plot
+into multiple subplots based on a variable included in the dataset. We will use it to
+make a time series plot for each species:
+
+```{r first-facet}
+surveys_abun_species %>%
+    group_by(year, species) %>%
+    tally() %>% 
+    ggplot(aes(x = year, y = n)) + #
+        geom_line() +
+        facet_wrap(~ species)
+```
+
+Now we would like to split the line in each plot by the sex of each individual
+measured. To do that we need to make counts in the data frame grouped by `year`,
+`species_id`, and `sex`:
+
+```{r}
+surveys_abun_species %>%
+    group_by(year, species, sex) %>%
+    tally()
+```
+
+We can make the faceted plot by splitting further by sex using `color` (within a single plot):
+
+```{r facet-by-species-and-sex}
+surveys_abun_species %>%
+    group_by(year, species, sex) %>%
+    tally() %>% 
+    ggplot(aes(x = year, y = n, color = sex)) +
+        geom_line() +
+        facet_wrap(~ species)
+```
+
+There are several observations where no sex was recorded. In this case, we are not really interested in the observations of unknown sex, so we can filter out those values.
+
+```{r}
+surveys_abun_species %>%
+    filter(!is.na(sex)) %>% 
+    group_by(year, species, sex) %>%
+    tally() %>% 
+    ggplot(aes(x = year, y = n, color = sex)) +
+        geom_line() +
+        facet_wrap(~ species)
+```
+
+It is possible specify exactly which colors to use, to avoid those that are hard
+to distinguish. We can also change the thickness of the lines, and adjust the
+xlabels so that they don't overlap. This will be discussed in much more detail
+in the data visualization lecture.
+
+```{r}
+color_names <- c('black', 'orange')
+
+surveys_abun_species %>%
+    filter(!is.na(sex)) %>% 
+    group_by(year, species, sex) %>%
+    tally() %>% 
+    ggplot(aes(x = year, y = n, color = sex)) +
+        geom_line(size = 1) +
+        scale_color_manual(values = color_names) +
+        facet_wrap(~ species) +
+        theme_bw() +
+        theme(text = element_text(size=12),
+               axis.text.x = element_text(angle=30, hjust=1)) 
+```
+
+
+> #### Challenge
+>
+> Use the filtered dataframe for all of these questions.
+> 1. Remember this histogram colored according to each species? Starting from that
```

Comment:
Space above this line.

Reply:
`Remember the histogram colored according to each species? Starting from that code, how could we separate each species into its own subplot?`